### PR TITLE
WIP: Refactoring of calculator code

### PR DESCRIPTION
**Pull request by maarten551**
It's such a mess now.

@eclipticrick can you please look at the changes? How much are they going against the rules of React?

### CODE REVIEW

**Comment by maarten551** [comment]
This is just the start. However, I want to know if this is correct by your standards first @eclipticrick :)

File: src/containers/Calculator/WindowMenu/Logic/Action/AboutAction.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export class AboutAction extends Action {
+    public execute(environment: Calculator): void {
+        const app = applications.find(context => context.id === ApplicationId.POPUP_INFO);
+        app.window.title = 'About Calculator';
+        app.window.children = (
+            <div className={popupClasses.padding}>
+                <img alt='about' className={popupClasses.left} src={infoIcon}/>
+                <div className={popupClasses.info}>
+                    Version 0.0.2<br/>
+                    © Wesley Veenendaal & Maarten Bobbeldijk :)
```
Comment (eclipticrick) [inline src/containers/Calculator/WindowMenu/Logic/Action/AboutAction.tsx:17]: @maarten551 nice x)

**Comment by maarten551** [comment]
@eclipticrick I only wanted some feedback on my code 😢

**Comment by eclipticrick** [comment]
> @eclipticrick I only wanted some feedback on my code 😢

'+1'
There's your feedback!

**Pull request merged**